types/timestamp.go

```diff
@@ -31,3 +31,17 @@ func (ts TimestampSlice) Less(i, j int) bool {
 func (ts TimestampSlice) Swap(i, j int) {
 	ts[i], ts[j] = ts[j], ts[i]
 }
+
+// Clock allows clients to retrieve the current time.
+type Clock interface {
+	Now() Timestamp
+}
+
+// StdClock is an implementation of Clock that retrieves the current time using
+// the system time.
+type StdClock struct{}
+
+// Now retrieves the current timestamp.
+func (c StdClock) Now() Timestamp {
+	return Timestamp(time.Now().Unix())
+}
```

Comment:
Yeah, that's a good point. We should get consistent file layouts.

I'll clarify that I don't believe it's useful to think about interfaces in terms of "mocking interfaces" or "non-mocking interfaces" because I don't think there is a distinction. I don't think it's possible for us to define the term "mocking interface" in a way that doesn't just mean the same as "interface." I think it's good design to have types call one another via interfaces rather than static types regardless of whether you create mock implementations of those interfaces for testing.

My suggestion:

```
// everything in your proposal up to the type ( ... ) section

type (
    // aliases

    // interfaces

    // structs
)

// functions of structs

// free functions (functions not tied to a struct)
```